src/base/text/smartstring.cpp

```diff
@@ -51,7 +51,8 @@ std::vector<std::string> SmartString::split(const std::string &str,
 std::string SmartString::fromNumber(double value,
     NumberFormat format,
     size_t maxFractionDigits,
-    const NumberScale &numberScale)
+    const NumberScale &numberScale,
+    const std::string &unit)
```

Comment:
If it has a unit as well, then it is no longer a number->string converter, but rather a physical value -> string formatter. The name should reflect this change.